src/views/EventsAttended/index.js

```diff
@@ -8,89 +8,81 @@ import { gordonColors } from 'theme';
 
 import { List, Grid, Button, Typography } from '@material-ui/core';
 
-export default class EventsAttended extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      events: [],
-      loading: true,
-    };
-  }
-  componentDidMount() {
-    if (this.props.authentication) {
-      this.loadEvents();
-    }
-  }
+const EventsAttended = (props) => {
+  const [events, setEvents] = useState([]);
+  const [loading, setLoading] = useState(true);
 
-  async loadEvents() {
-    this.setState({ loading: true });
-    const events = await event.getAttendedChapelEvents();
-    this.setState({ events, loading: false });
-  }
-
-  render() {
-    let content;
-
-    const style = {
-      button: {
-        background: gordonColors.primary.cyan,
-        color: 'white',
-      },
+  useEffect(() => {
+    const loadEvents = async () => {
+      if (props.authentication) {
```

Comment:
```suggestion
      setLoading(true);
      if (props.authentication) {
```
In other places where we need to load things, we typically set loading to true even if the initial value is true just to make it more clear. It's up to you whether to stick with that though; this should work fine.

Reply:
This would make sense for many of the other files that use useEffect, however, as Cam pointed out to me, events attended will only be ever loaded once, when the page is rendered. So the useState function declared at the beginning is the only time we have to set loading to true since we never update the page based on other information.

Reply:
Yes, that's correct. It's more for consistency's sake; in the carousel file, for example, you DO set a loading property to true before loading, even though you initialize it to true and it is only loaded once. It is up to you what you decide on, though.